sphinx/ext/autosummary/generate.py

```diff
@@ -221,12 +217,13 @@ def find_autosummary_in_files(filenames):
     """
     documented = []
     for filename in filenames:
-        f = open(filename, 'r')
-        lines = f.read().splitlines()
-        documented.extend(find_autosummary_in_lines(lines, filename=filename))
-        f.close()
+        with open(filename, 'r', encoding='utf-8') as f:
```

Comment:
This is the change I'm not 100% sure about. I was having a problem on `master` on Windows compiling some `.rst` files, hitting an encoding error. When I explicitly set the encoding to use when reading the file, the error goes away. Is this likely to be general enough for general use...?